File: client-scalaz/src/main/scala/org/openapitools/client/api/PathingResponse.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime


import PathingResponse._

case class PathingResponse (
  start: Option[NodeRequest],
end: Option[NodeRequest],
waypoints: Option[List[NodeRequest]],
exclusions: Option[List[NodeRequest]],
width: Option[Integer],
height: Option[Integer],
southwest: Option[NodeRequest],
northeast: Option[NodeRequest],
metersPerX: Option[Double],
metersPerY: Option[Double],
`path`: Option[List[NodeRequest]],
pathCount: Option[Integer],
directions: Option[List[DirectionResponse]],
directionCount: Option[Integer],
swcalibration: Option[NodeRequest],
calibrationBearings: Option[Double],
necalibration: Option[NodeRequest])

object PathingResponse {
  import DateTimeCodecs._

  implicit val PathingResponseCodecJson: CodecJson[PathingResponse] = CodecJson.derive[PathingResponse]
  implicit val PathingResponseDecoder: EntityDecoder[PathingResponse] = jsonOf[PathingResponse]
  implicit val PathingResponseEncoder: EntityEncoder[PathingResponse] = jsonEncoderOf[PathingResponse]
}

File: client-clojure/src/sirqul_io_t_platform/specs/pathing_response.clj
(ns sirqul-io-t-platform.specs.pathing-response
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            [sirqul-io-t-platform.specs.node-request :refer :all]
            [sirqul-io-t-platform.specs.node-request :refer :all]
            [sirqul-io-t-platform.specs.node-request :refer :all]
            [sirqul-io-t-platform.specs.node-request :refer :all]
            [sirqul-io-t-platform.specs.node-request :refer :all]
            [sirqul-io-t-platform.specs.node-request :refer :all]
            [sirqul-io-t-platform.specs.node-request :refer :all]
            [sirqul-io-t-platform.specs.direction-response :refer :all]
            [sirqul-io-t-platform.specs.node-request :refer :all]
            [sirqul-io-t-platform.specs.node-request :refer :all]
            )
  (:import (java.io File)))


(def pathing-response-data
  {
   (ds/opt :start) node-request-spec
   (ds/opt :end) node-request-spec
   (ds/opt :waypoints) (s/coll-of node-request-spec)
   (ds/opt :exclusions) (s/coll-of node-request-spec)
   (ds/opt :width) int?
   (ds/opt :height) int?
   (ds/opt :southwest) node-request-spec
   (ds/opt :northeast) node-request-spec
   (ds/opt :metersPerX) float?
   (ds/opt :metersPerY) float?
   (ds/opt :path) (s/coll-of node-request-spec)
   (ds/opt :pathCount) int?
   (ds/opt :directions) (s/coll-of direction-response-spec)
   (ds/opt :directionCount) int?
   (ds/opt :swcalibration) node-request-spec
   (ds/opt :calibrationBearings) float?
   (ds/opt :necalibration) node-request-spec
   })

(def pathing-response-spec
  (ds/spec
    {:name ::pathing-response
     :spec pathing-response-data}))
